File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30Strategy/src/main/java/ec/edu/espe/controller/SortingContext.java

package ec.edu.espe.controller;

/**
 *
 * @author MSI-PULSE
 */
public class SortingContext {
    private SortingStrategy strategy;

    public void setSortStrategy(SortingStrategy strategy) {
        this.strategy = strategy;
    }

    public void sort(int[] numbers) {
        strategy.sort(numbers);
    }

    public String getSortStrategyName() {
        return strategy.getStrategyName();
    }
}

File: exams/gavilanez/u3/ExamP3(2)/ec_edu_espe_examp3_model/SortingContext.py
from ec_edu_espe_examp3_controller.SortingStrategy import SortingStrategy

class SortingContext:
    def __init__(self):
        self._strategy = None

    def set_sort_strategy(self, strategy: SortingStrategy):
        self._strategy = strategy

    def sort(self, numbers):
        return self._strategy.sort(numbers)
